index.js

```diff
@@ -50,7 +50,7 @@ exports = module.exports = (function () {
 
     function getHarness(opts) {
         if (!opts) opts = {};
-        opts.autoclose = !canEmitExit;
+        opts.autoclose = opts.autoclose || !canEmitExit;
```

Comment:
so if `autoclose` is `true`, this will be `true`, but if it's `false`, this won't be guaranteed to be false.

should this instead be?:
```suggestion
        opts.autoclose = typeof opts.autoclose === 'boolean' ? opts.autoclose : !canEmitExit;
```
?
or instead should it be:
```suggestion
        opts.autoclose = canEmitExit ? opts.autoclose : false;
```

Reply:
`canEmitExit` is based on the environment, but `opts.autoclose` is meant to close the stream before exit whatever the environment be. (The first suggestion is what I'am going for)

What are your thoughts on `canEmitExit`? Should we leave it alone or close the stream once the tests are `done` whatever the environment maybe?

Reply:
I'm a bit unclear still, streams aren't where my knowledge lies :-) can you explain a bit more?

Reply:
`canEmitExit` is used to know if `tape` is running in browser or in a process like node. If inside a browser the stream that spits out the results/output is closed when the "results end", this is done when the `done` event is fired on the results.

But in environments like node the steam is closed when the process ends. This is to facilitate the use of `tape test/*.js` usage (what I think is). But when this is used the last 3 lines for the output (the one which shows the aggregated resuls) which I was trying won't be spit out when I use `tape` as a module. When run through `tape test/*.js` the stream spits those 3 lines to `stdout`, thus the output is preserved. But when required as a module and a stream is created manually, the output is contained inside the process, when it ends the output is gone because the stream is just a variable that died without spitting it to `stdout`. This is delayed by the `resumer` package. It waits to put the chunks out onto the target stream.

My actual suggestion is to actually remove `canEmitExit` totally and to just close the stream right away. But I think it might cause other problems.

Reply:
What other problems would it solve?

Reply:
Are there circumstances where someone might want the process to exit immediately (initiated by user code) and tape would still need to suppress the final output?

Reply:
> What other problems would it cause?

I don't think it would cause any

> Are there circumstances where someone might want the process to exit immediately (initiated by user code) and tape would still need to suppress the final output?

I don't think so, if the process exits normally then all the other tests are reported as "exited without ending", when it ends abnormally then nothing extra is printed, the whole process just quits. In both cases the final output is not shown 😟

Reply:
If no tests break, and there's no obvious problems caused by removing `canEmitExit`, we might as well remove it?

Reply:
Yes, and also I've spotted a few unused variables lying around that have a purpose but are not used at all just like `canEmitExit`. I wonder, why actually they were written.

Reply:
¯\\\_(ツ)_/¯ it predates my involvement in the project.

Reply:
I'll see what I can do. I'll try to remove `canEmitExit`, and test it, Add tests for the current situation and remove the `opts.autoclose`

Reply:
I had a good sleep and after looking around it looks like I was wrong. `canEmitExit` is there so that the process exists with the correct exit code. Looks like we need the explicit `autoclose` for that one specific purpose.

Reply:
Reporters like `tap-mocha-reporter` work fine without `autoclose`, it consumes the stream quickly and directly spits it to `stdout` only `tap-spec` has this problem because it uses it's own `tap-out` parser which tries to read the stream in a different way.

Reply:
Alas!! I've found a way where `tap-out` could work just like `tap-mocha-reporter`. We don't need to implement `autoclose` anymore, it depends on the reporter now.

We can ignore this PR now I think 😀

Reply:
Thanks for the update!